index.rst

```diff
@@ -2875,8 +2898,51 @@ query, and the impact is amortized among all disks on all workers.
 For discussion about the performance of the existing prototype, refer to
 :ref:`demo-shared-scans`.
 
+.. _shared-scan-memory_management
+
+Memory management
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To minimize system paging when multiple threads are scanning the same
+table, we implemented a memory manager called `memman`_. When a shared
+scan is about to commence, the shared scan scheduler informs `memman`_
+about the tables the query will be using and how important it is to
+keep those tables in memory during the course of the query. When
+directed to keep teh tables in memory, `memman`_ opens each data base
+table file, maps it into memory, and then locks the pages to prevent
+the kernel from stealing the pages for other uses. Thus, once a file
+page is faulted in, it stays in memory and allows other threads to
+scan the contents of the page without incurring additional page faults.
+Once the the shared scan of the table completes, `memman`_ is told
+that the tables no longer need to remain in memory. `memman`_ frees
+up the pages by unlocking them and deleting the mapping.
+
+This type of management is necessary because the prime paging pool
+to satisfy system paging requirements is the set of unlocked pages
```

Comment:
This type of management is necessary to satisfy system paging requirements because the prime paging pool
 is the set of unlocked pages